packages/sit-onyx/src/components/OnyxButton/OnyxButton.vue

```diff
@@ -0,0 +1,157 @@
+<script lang="ts" setup>
+import type { ButtonProps } from "./types";
+const props = withDefaults(defineProps<ButtonProps>(), {
+  label: "",
+  disabled: false,
+  type: "button",
+  color: "primary",
+  variant: "default",
+});
+
+const emit = defineEmits<{
+  /**
+   * Emitted when the button is clicked
+   */
+  click: [];
+}>();
+</script>
+
+<template>
+  <button
+    class="onyx-button"
+    :class="[`onyx-button--${props.color}`, `onyx-button--${props.variant}`]"
+    :disabled="props.disabled"
+    @click="emit('click')"
+  >
+    <span class="onyx-button__label">{{ props.label }}</span>
+  </button>
+</template>
+
+<style lang="scss">
+.onyx-button {
+  display: inline-flex;
```

Comment:
```suggestion
  display: flex;
```
I would prefer a button to not be an inline element, as it is not indended to be used inline.